src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -351,12 +363,241 @@ protected Expr createTransportSettingsInitExpr(
+        Variable.builder().setType(protoMethod.outputType()).setName("response").build());
+
+    List<Statement> createBody = new ArrayList<Statement>(4);
+
+    // Generate opName
+    TypeNode stringBuilderType =
+        TypeNode.withReference(ConcreteReference.withClazz(StringBuilder.class));
+    VariableExpr opNameVarExpr =
+        VariableExpr.withVariable(
+            Variable.builder().setType(stringBuilderType).setName("opName").build());
+    MethodInvocationExpr getId =
+        MethodInvocationExpr.builder()
+            .setMethodName("getId")
+            .setExprReferenceExpr(responseVarExpr).build();
+    Expr opNameObjectExpr = NewObjectExpr.builder()
+        .setType(stringBuilderType)
+        .setArguments(getId)
+        .build();
+    AssignmentExpr opNameAssignExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(opNameVarExpr.toBuilder().setIsDecl(true).build())
+            .setValueExpr(opNameObjectExpr)
+            .build();
+    createBody.add(ExprStatement.withExpr(opNameAssignExpr));
+
+    // Generate changes opName
+    MethodInvocationExpr requestGetProjectExpr = MethodInvocationExpr.builder()
+        .setMethodName("getProject")
+        .setExprReferenceExpr(requestVarExpr)
+        .build();
+    ValueExpr colonValueExpr =
+        ValueExpr.builder().setValue(StringObjectValue.builder().setValue(":").build()).build();
+    MethodInvocationExpr opNameAppendColonProjectExpr = MethodInvocationExpr.builder()
+        .setMethodName("append")
+        .setArguments(colonValueExpr)
+        .setExprReferenceExpr(opNameVarExpr)
+        .build();
+    opNameAppendColonProjectExpr = methodMaker.apply(
+        "append", Collections.singletonList(requestGetProjectExpr))
+        .apply(opNameAppendColonProjectExpr);
+    createBody.add(ExprStatement.withExpr(opNameAppendColonProjectExpr));
+
+    // Generate changes to opName
+    MethodInvocationExpr requestGetRegionExpr = MethodInvocationExpr.builder()
+        .setMethodName("getRegion")
+        .setExprReferenceExpr(requestVarExpr)
+        .build();
+    MethodInvocationExpr opNameAppendColonRegionExpr = MethodInvocationExpr.builder()
+        .setMethodName("append")
+        .setArguments(colonValueExpr)
+        .setExprReferenceExpr(opNameVarExpr)
+        .build();
+    opNameAppendColonRegionExpr = methodMaker.apply(
+        "append", Collections.singletonList(requestGetRegionExpr))
+        .apply(opNameAppendColonRegionExpr);
+    createBody.add(ExprStatement.withExpr(opNameAppendColonRegionExpr));
+
+    // Generate check status expression
+    MethodInvocationExpr getStatusExpr = MethodInvocationExpr.builder()
+        .setExprReferenceExpr(responseVarExpr)
+        .setMethodName("getStatus")
+        .build();
+    TypeNode statusType = TypeNode.withReference(
+        VaporReference.builder()
+            .setName("Status")
+            .setPakkage("com.google.cloud.compute.v1")
```

Comment:
For every line of code, which has now compute hardcoded, add a corresponding TODO (that will be easy to search for , when we will be replacing the hardcoded parts).